lang/cs/Org.Apache.REEF.IMRU/OnREEF/CheckpointHandler/CheckpointConfigurationModule.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using Org.Apache.REEF.IMRU.API;
+using Org.Apache.REEF.IMRU.OnREEF.IMRUTasks;
+using Org.Apache.REEF.Tang.Formats;
+using Org.Apache.REEF.Tang.Util;
+using Org.Apache.REEF.Wake.Remote;
+
+namespace Org.Apache.REEF.IMRU.OnREEF.CheckpointHandler
+{
+    public sealed class CheckpointConfigurationModule : ConfigurationModuleBuilder
+    {
+        public static readonly RequiredParameter<string> CheckpointFile = new RequiredParameter<string>();
```

Comment:
How does this work? Is the checkpoint assumed to be written to a local file? How would that survive the crash of the Update Task?

Reply:
Client specify the checkpoint file path, it is usually a path in the remote storage and injected file system have access to it. The state files will be written to remote file path location.